compiler/noirc_frontend/src/hir/comptime/scan.rs

```diff
@@ -0,0 +1,206 @@
+//! This module is for the scanning of the Hir by the interpreter.
+//! In this initial step, the Hir is scanned for `CompTime` nodes
+//! without actually executing anything until such a node is found.
+//! Once such a node is found, the interpreter will call the relevant
+//! evaluate method on that node type, insert the result into the Ast,
+//! and continue scanning the rest of the program.
+//!
+//! Since it mostly just needs to recur on the Hir looking for CompTime
+//! nodes, this pass is fairly simple. The only thing it really needs to
+//! ensure to do is to push and pop scopes on the interpreter as needed
+//! so that any variables defined within e.g. an `if` statement containing
+//! a `CompTime` block aren't accessible outside of the `if`.
+use crate::{
+    hir_def::{
+        expr::{
+            HirArrayLiteral, HirBlockExpression, HirCallExpression, HirConstructorExpression,
+            HirIfExpression, HirIndexExpression, HirInfixExpression, HirLambda,
+            HirMethodCallExpression,
+        },
+        stmt::HirForStatement,
+    },
+    macros_api::{HirExpression, HirLiteral, HirStatement},
+    node_interner::{ExprId, FuncId, StmtId},
+};
+
+use super::{
+    errors::{IResult, InterpreterError},
+    interpreter::Interpreter,
+};
+
+#[allow(dead_code)]
+impl<'interner> Interpreter<'interner> {
+    /// Scan through a function, evaluating any CompTime nodes found.
+    /// These nodes will be modified in place, replaced with the
+    /// result of their evaluation.
+    pub fn scan_function(&mut self, function: FuncId) {
+        let function = self.interner.function(&function);
+
+        let state = self.enter_function();
+        self.scan_expression(function.as_expr()).unwrap();
```

Comment:
We just unwrap here? Don't we want to thread the interpreter error up?

Reply:
This is a stub for now until error handling is added into another PR. I'll want to turn these into CompilerErrors but want to save that conversion into another PR. I'll move the unwrap to where it is called after type checking and add a comment explaining this for now.

Reply:
Ok that works!